hooks/cmd/install/install.go

```diff
@@ -0,0 +1,225 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import 	(
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+        "log/syslog"
+	"os"
+	"strconv"
+
+	"github.com/canonical/app-service-configurable/hooks"
+)
+
+var log syslog.Writer
+
+func handleDirs() error {
+	dataConfP := fmt.Sprintf("%s/config/res", hooks.SnapDataE)
+	snapConfP := fmt.Sprintf("%s/config/res", hooks.SnapE)
+
+	// install all the config files from $SNAP/config/res/ into $SNAP_DATA/config/res/,
+	// but if files already exist, don't over-write
+	log.Info(fmt.Sprintf("edgex-asc:install creating: %s", dataConfP))
+	err := os.MkdirAll(dataConfP, 0755)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Can't make %s - %v", dataConfP, err))
```

Comment:
this can be `fmt.Errorf`:

```suggestion
		return fmt.Errorf("Can't make %s - %v", dataConfP, err)
```

also since you are using go 1.15, you can do the fancy thing and "wrap" your errors using the %w error verb:

```suggestion
		return fmt.Errorf("Can't make %s - %w", dataConfP, err)
```

you probably don't need this now, but this lets you inspect the returned error with i.e. os.IsNotExist() even if you add special text on top of that with fmt.Errorf, whereas if you just do `%v` for the error verb, then that information is lost and os.IsNotExist will return false, even if the original error was indeed os.ErrNotExist

Reply:
also little nitpick is that error messages shouldn't be capitalized in Go